vcd/resource_vcd_rde_type.go

```diff
@@ -0,0 +1,344 @@
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`(?i)^[a-z0-9_-]+$`), "only alphanumeric characters, underscores and hyphens allowed"),
+				Description:  "A unique namespace associated with the Runtime Defined Entity Type. Combination of `vendor`, `nss` and `version` must be unique",
+			},
+			"version": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The version of the Runtime Defined Entity Type. The version string must follow semantic versioning rules. Combination of `vendor`, `nss` and `version` must be unique",
+			},
+			"interface_ids": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "Set of Defined Interface URNs that this Runtime Defined Entity Type is referenced by",
+			},
+			"schema_url": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "URL that should point to a JSON-Schema valid definition file of the Runtime Defined Entity Type",
+				ExactlyOneOf: []string{"schema_url", "schema"},
+			},
+			"schema": {
+				Type:                  schema.TypeString,
+				Optional:              true,
+				Computed:              true,
+				Description:           "The JSON-Schema valid definition of the Runtime Defined Entity Type",
+				ExactlyOneOf:          []string{"schema_url", "schema"},
+				DiffSuppressFunc:      hasJsonValueChanged,
+				DiffSuppressOnRefresh: true,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The description of the Runtime Defined Entity Type",
+			},
+			"external_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "An external entity's ID that this definition may apply to",
+			},
+			"inherited_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "To be used when creating a new version of a Runtime Defined Entity Type. Specifies the version of the type that will be the template for the authorization configuration of the new version." +
+					"The Type ACLs and the access requirements of the Type Behaviors of the new version will be copied from those of the inherited version." +
+					"If not set, then the new type version will not inherit another version and will have the default authorization settings, just like the first version of a new type",
+			},
+			"readonly": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "True if the Runtime Defined Entity Type cannot be modified",
+			},
+		},
+	}
+}
+
+// hasJsonValueChanged tells Terraform whether the JSON schema set in HCL configuration (which can have whatever identation and other quirks)
+// matches the obtained JSON from VCD.
+func hasJsonValueChanged(key, oldValue, newValue string, _ *schema.ResourceData) bool {
```

Comment:
I'm not sure how it's done in other files, but would it make sense to have it in a more general file like `structure.go`?